neo3/wallet/account.py

```diff
@@ -0,0 +1,258 @@
+        encrypted = decoded_key[-32:]
+
+        pwd_normalized = bytes(unicodedata.normalize('NFC', passphrase), 'utf-8')
+        derived = hashlib.scrypt(password=pwd_normalized, salt=address_checksum,
+                                 n=16384,
+                                 r=8,
+                                 p=8,
+                                 dklen=64)
+
+        derived1 = derived[:32]
+        derived2 = derived[32:]
+
+        cipher = AES.new(derived2, AES.MODE_ECB)
+        decrypted = cipher.decrypt(encrypted)
+        private_key = Account.xor_bytes(decrypted, derived1)
+
+        # Now check that the address hashes match. If they don't, the password was wrong.
+        key_pair = KeyPair(private_key=private_key)
+        script_hash = to_script_hash(contracts.Contract.create_signature_redeemscript(key_pair.public_key))
+        address = Account.script_hash_to_address(script_hash)
+        first_hash = hashlib.sha256(address.encode("utf-8")).digest()
+        second_hash = hashlib.sha256(first_hash).digest()
+        checksum = second_hash[:4]
+        if checksum != address_checksum:
+            raise ValueError("Wrong passphrase")
+
+        return private_key
+
+    @staticmethod
+    def private_key_to_nep2(private_key: bytes, passphrase: str) -> bytes:
+        """
+        Encrypt a private key into a nep2 key.
+
+        Args:
+            private_key: the key that will be encrypt.
+            passphrase: the password to encrypt the nep2 key.
+
+        Returns:
+            the encrypted nep2 key.
+        """
+        key_pair = KeyPair(private_key=private_key)
+        script_hash = to_script_hash(contracts.Contract.create_signature_redeemscript(key_pair.public_key))
+        address = Account.script_hash_to_address(script_hash)
+        # NEP2 checksum: hash the address twice and get the first 4 bytes
+        first_hash = hashlib.sha256(address.encode("utf-8")).digest()
+        second_hash = hashlib.sha256(first_hash).digest()
+        checksum = second_hash[:4]
+
+        pwd_normalized = bytes(unicodedata.normalize('NFC', passphrase), 'utf-8')
+        derived = hashlib.scrypt(password=pwd_normalized, salt=checksum,
+                                 n=16384,
+                                 r=8,
+                                 p=8,
+                                 dklen=64)
+
+        derived1 = derived[:32]
+        derived2 = derived[32:]
+
+        xor_ed = Account.xor_bytes(bytes(private_key), derived1)
+        cipher = AES.new(derived2, AES.MODE_ECB)
+        encrypted = cipher.encrypt(xor_ed)
+
+        nep2 = bytearray()
+        nep2.extend(NEP_HEADER)
+        nep2.extend(NEP_FLAG)
+        nep2.extend(checksum)
+        nep2.extend(encrypted)
+
+        # Finally, encode with Base58Check
+        encoded_nep2 = base58.b58encode_check(bytes(nep2))
+
+        return encoded_nep2
+
+    @staticmethod
+    def xor_bytes(a: bytes, b: bytes) -> bytes:
```

Comment:
I'm leaning towards making this a private method as it is only used as a helper internally.